src/server/server.rs

```diff
@@ -344,6 +396,33 @@ where
         Ok(())
     }
 
+    pub fn pause(&mut self) -> Result<()> {
+        let start = Instant::now();
+        // Prepare the builder for resume grpc server. And if the builder cannot be
+        // created, then pause will be skipped.
+        let builder = Either::Left(self.builder_factory.create_builder(self.env.clone())?);
+        if let Some(Either::Right(server)) = self.builder_or_server.take() {
+            drop(server);
+        }
+        self.health_service
+            .set_serving_status("", ServingStatus::NotServing);
+        self.builder_or_server = Some(builder);
+        info!("paused the grpc server"; "spent_time" => ?start.elapsed(),);
+        Ok(())
+    }
+
+    pub fn resume(&mut self) -> Result<()> {
+        if let Some(builder) = self.builder_or_server.as_ref() {
+            let start = Instant::now();
+            assert!(builder.is_left());
+            self.build_and_bind()?;
```

Comment:
Does caller retry if it fails to bind ports?

Reply:
Yep, it will warn with `failed to resume the grpc server` logging if it fails. And the user can find it to repeat the operation again.